app/services/candidate_interface/courses_recommender.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+module CandidateInterface
+  class CoursesRecommender
+    include Rails.application.routes.url_helpers
+
+    def self.recommended_courses_url(candidate:, locatable: nil)
+      new(candidate:, locatable:).recommended_courses_url
+    end
+
+    # @param [Provider, Site, ApplicationForm] locatable
+    def initialize(candidate:, locatable: nil)
+      @candidate = candidate
+      @locatable = locatable
+    end
+
+    def recommended_courses_url
+      find_url_with_query_params
+    end
+
+  private
+
+    attr_reader :candidate, :locatable
+
+    def find_url_with_query_params
+      uri = URI("#{find_url}results")
+      uri.query = query_parameters.to_query unless query_parameters.empty?
```

Comment:
Is it worth showing the `Recommended courses` link if we don't have query params here?

As we would send the user to all the courses in find, 10k course it seems, it's not really a recommendation.

Reply:
I flip-flopped on this all the way through the build - I'll go back and add the logic to remove the URL if we're not really recommending anything